rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -448,86 +449,95 @@ void game_list_frame::Refresh(const bool fromDrive, const bool scrollAfter)
-			}
+				QString serial = qstr(game.serial);
+				m_notes[serial] = m_gui_settings->GetValue(gui::notes, serial, "").toString();
+				m_titles[serial] = m_gui_settings->GetValue(gui::titles, serial, "").toString().simplified();
+				serials.insert(serial);
+
+				auto cat = category::cat_boot.find(game.category);
+				if (cat != category::cat_boot.end())
+				{
+					game.icon_path = sfo_dir + "/ICON0.PNG";
+					game.category = sstr(cat->second);
+				}
+				else if ((cat = category::cat_data.find(game.category)) != category::cat_data.end())
+				{
+					game.icon_path = sfo_dir + "/ICON0.PNG";
+					game.category = sstr(cat->second);
+				}
+				else if (game.category == cat_unknown)
+				{
+					game.icon_path = sfo_dir + "/ICON0.PNG";
+				}
+				else
+				{
+					game.icon_path = sfo_dir + "/ICON0.PNG";
+					game.category = sstr(category::other);
+				}
 
-			// Load ICON0.PNG
-			QPixmap icon;
+				// Load ICON0.PNG
+				QPixmap icon;
 
-			if (game.icon_path.empty() || !icon.load(qstr(game.icon_path)))
-			{
-				LOG_WARNING(GENERAL, "Could not load image from path %s", sstr(QDir(qstr(game.icon_path)).absolutePath()));
-			}
+				if (game.icon_path.empty() || !icon.load(qstr(game.icon_path)))
+				{
+					LOG_WARNING(GENERAL, "Could not load image from path %s", sstr(QDir(qstr(game.icon_path)).absolutePath()));
+				}
 
-			const auto compat = m_game_compat->GetCompatibility(game.serial);
+				const auto compat = m_game_compat->GetCompatibility(game.serial);
 
-			const bool hasCustomConfig = fs::is_file(Emulator::GetCustomConfigPath(game.serial)) || fs::is_file(Emulator::GetCustomConfigPath(game.serial, true));
-			const bool hasCustomPadConfig = fs::is_file(Emulator::GetCustomInputConfigPath(game.serial));
+				const bool hasCustomConfig = fs::is_file(Emulator::GetCustomConfigPath(game.serial)) || fs::is_file(Emulator::GetCustomConfigPath(game.serial, true));
+				const bool hasCustomPadConfig = fs::is_file(Emulator::GetCustomInputConfigPath(game.serial));
 
-			const QColor color = getGridCompatibilityColor(compat.color);
-			const QPixmap pxmap = PaintedPixmap(icon, hasCustomConfig, hasCustomPadConfig, color);
+				const QColor color = getGridCompatibilityColor(compat.color);
+				const QPixmap pxmap = PaintedPixmap(icon, hasCustomConfig, hasCustomPadConfig, color);
 
-			m_game_data.push_back(game_info(new gui_game_info{game, compat, icon, pxmap, hasCustomConfig, hasCustomPadConfig}));
-		}
-		catch (const std::exception& e)
-		{
-			LOG_FATAL(GENERAL, "Failed to update game list at %s\n%s thrown: %s", dir, typeid(e).name(), e.what());
-			continue;
-			// Blame MSVC for double }}
-		}}
+				m_game_data.push_back(game_info(new gui_game_info{ game, compat, icon, pxmap, hasCustomConfig, hasCustomPadConfig }));
+			}
+			catch (const std::exception& e)
+			{
+				LOG_FATAL(GENERAL, "Failed to update game list at %s\n%s thrown: %s", dir, typeid(e).name(), e.what());
+				return;
+				// Blame MSVC for double }}
```

Comment:
Random nitpick: comment isn't applicable anymore. Besides, it wasn't the compiler's fault to begin with :^)

Reply:
It was

Reply:
```cpp
for (const auto& dir : path_list) { try
{
// ...
}
catch (const std::exception& e)
{
	LOG_FATAL(GENERAL, "Failed to update game list at %s\n%s thrown: %s", dir, typeid(e).name(), e.what());
	continue;
	// Blame MSVC for double }}
} /* ends try-catch */ } /* ends for-loop */
```
Are you still sure about that?

Reply:
Yes

Reply:
I don't think you understand the comment

Reply:
It's all good ;)

Reply:
Not really a bug, lol. I think other compilers can run the try catch expressions in a loop without the {}. At least that's how I remember the explanation

Reply:
Ah, I see - you want to do `for (...) try { ... }`. And the compiler doesn't get it? Seriously?

Reply:
sigh.  not me. the person who wrote that comment

Reply:
Well, you get the point.